misk/src/main/kotlin/misk/environment/EnvironmentModule.kt

```diff
@@ -1,10 +1,26 @@
 package misk.environment
 
 import com.google.inject.AbstractModule
+import misk.logging.getLogger
 
-class EnvironmentModule(private val environment: Environment) : AbstractModule() {
+private val logger = getLogger<EnvironmentModule>()
+
+class EnvironmentModule(val environment: Environment) : AbstractModule() {
     override fun configure() {
         bind(Environment::class.java).toInstance(environment)
     }
 
+    companion object {
+        @JvmStatic
+        fun fromEnvironmentVariable(): EnvironmentModule {
+            val environmentName = System.getenv("ENV")
+            val environment = if (environmentName != null) {
+                Environment.valueOf(environmentName.toUpperCase())
```

Comment:
Not a fan of converting it to upper case here actually. We want our users to just use the correct case don’t we? I think fewer clever behaviors is better.

Independently, you never want `toUpperCase()`; that API is very toxic and should never be used. Instead use `toUpperCase(Locale.US)`. Otherwise `staging` is converted to `STAGİNG` in Turkey.

Reply:
On second thought, I think you're right. We shouldn't try to be super clever, since that's what confuses people when things don't work. We want to enforce the correct usage rather than trying to consider all possible edge cases. Will fix in a followup.